src/Symfony/Component/Lock/Store/MongoDbStore.php

```diff
@@ -0,0 +1,421 @@
+     */
+    public function __construct(Client $mongo, array $options, float $initialTtl = 300.0)
+    {
+        $this->mongo = $mongo;
+
+        $this->initialTtl = $initialTtl;
+
+        $this->options = array_merge(array(
+            'collection' => 'lock',
+            'gcProbablity' => 0.001,
+        ), $options);
+
+        if (!isset($this->options['database'])) {
+            throw new InvalidArgumentException(
+                'You must provide the "database" option for MongoDBStore'
+            );
+        }
+
+        if ($this->options['gcProbablity'] < 0.0 || $this->options['gcProbablity'] > 1.0) {
+            throw new InvalidArgumentException(sprintf(
+                '"%s" gcProbablity must be a float from 0.0 to 1.0, "%f" given.',
+                __METHOD__,
+                $this->options['gcProbablity']
+            ));
+        }
+
+        if ($this->initialTtl <= 0) {
+            throw new InvalidArgumentException(sprintf(
+                '%s() expects a strictly positive TTL. Got %d.',
+                __METHOD__,
+                $this->initialTtl
+            ));
+        }
+    }
+
+    /**
+     * Create a TTL index to automatically remove expired locks.
+     *
+     * If the gcProbablity option is set higher than 0.0 (defaults to 0.001);
+     * there is a chance this will be called on self::save().
+     *
+     * Otherwise; this should be called once manually during database setup.
+     *
+     * Alternatively the TTL index can be created manually on the database:
+     *
+     *  db.lock.ensureIndex(
+     *      { "expires_at": 1 },
+     *      { "expireAfterSeconds": 0 }
+     *  )
+     *
+     * Please note, expires_at is based on the application server. If the
+     * database time differs; a lock could be cleaned up before it has expired.
+     * To ensure locks don't expire prematurely; the lock TTL should be set
+     * with enough extra time to account for any clock drift between nodes.
+     *
+     * A TTL index MUST BE used on MongoDB 2.2+ to automatically clean up expired locks.
+     *
+     * @see http://docs.mongodb.org/manual/tutorial/expire-data/
+     *
+     * @return string The name of the created index
+     *
+     * @throws UnsupportedException          if options are not supported by the selected server
+     * @throws MongoInvalidArgumentException for parameter/option parsing errors
+     * @throws DriverRuntimeException        for other driver errors (e.g. connection errors)
+     */
+    public function createTtlIndex(int $expireAfterSeconds = 0): string
+    {
+        if ($this->getDatabaseVersion() < '2.2') {
+            throw new BadMethodCallException('Database version does not support TTL Indexes');
+        }
+
+        $keys = array(
+            'expires_at' => 1,
+        );
+
+        $options = array(
+            'expireAfterSeconds' => $expireAfterSeconds,
+        );
+
+        return $this->getCollection()->createIndex($keys, $options);
```

Comment:
What if the method is called several time (like you do in GC)? any impact on existing data or performances?

Reply:
`createIndex` is idempotent in MongoDb https://docs.mongodb.com/manual/indexes/#create-an-index

> The db.collection.createIndex method only creates an index if an index of the same specification does not already exist.